python/lsst/meas/astrom/matchOptimisticB.py

```diff
@@ -120,7 +131,12 @@ def hasCentroid(self, source):
     def isCleanSource(self, source):
         """Return True if the source has a valid centroid and is not near the edge
         """
-        return self.hasCentroid(source) and not source.get(self.edgeKey)
+        flux = source.get('base_PsfFlux_flux')
+        if flux > 0. :
+            mag = self.calib.getMagnitude(flux)
+        else :
+            mag = 99
+        return self.hasCentroid(source) and not source.get(self.edgeKey) and not source.get(self.crKey) and self.isIsolated(source) and not source.get(self.intKey) and source.get(self.extKey)<0.5 and mag<24 and not source.get(self.flagExtKey)
```

Comment:
Here is the selection criteria for the sources. Cutting on magnitude does not really help